### PR TITLE
[ Maintenance ] Consolidating worship-service-sensitive delta producer

### PR DESCRIPTION
**Pull request by Poltergeistz**
# Description
DL-5588

This PR extends the configuration of the maintainer and job-initiator for worship-services-sensitive deltas. It moves in config.json files environment variables of the docker-compose related to the services. It also removes the custom rule file to include it in the maintainer rule instead.

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

# Related services

- https://github.com/lblod/delta-producer-publication-graph-maintainer
- https://github.com/lblod/delta-producer-background-jobs-initiator
- https://github.com/lblod/delta-consumer

# How to test 

1. Add your `key` and `value` in your `app-digitaal-loket/config/delta-producer/publication-graph-maintainer/config.json` file

2. Launch Loket and OP (organisation-portal)
3. Check the logs of delta-publication-graph-maintainer if you have deltas already you wont need to create a new person and attach a worship position or mandate.
4. Depending of your delta-consumer service you'll need to set few variables 
```yml
  worship-services-sensitive-consumer:
    environment:
      DCR_DISABLE_INITIAL_SYNC: 'true'
      DCR_WAIT_FOR_INITIAL_SYNC: 'false'
      DCR_SECRET_KEY: "yourKey"
      DCR_SYNC_BASE_URL: 'http://endpoint-of-your-choice:90'
      DCR_SYNC_LOGIN_ENDPOINT: 'http://endpoint-of-your-choice:90/sync/worship-services-sensitive/login'
      DCR_START_FROM_DELTA_TIMESTAMP: "2023-11-09"
  file:
    environment:
      DISABLE_AUTOMATIC_SYNC: "true"
      DISABLE_INITIAL_SYNC: "true"
      FILES_ENDPOINT_BASE_URL: "http://endpoint-of-your-choice:90" # The endpoint of your choice (see later what to choose)
      SYNC_LOGIN_ENDPOINT: "http://endpoint-of-your-choice:90/sync/worship-services-sensitive/login"
      SECRET_KEY: "youKey"
```
5. Then log the worship-services-sensitive-consumer, it should consume the delta file created from loket.

# What to check

- N/A

# Links to other PR's

- N/A

# Notes

### CODE REVIEW

**Review by Poltergeistz — COMMENTED** [review]
Is there a reason that the route is duplicated ?

https://github.com/lblod/app-digitaal-loket/pull/552/commits/9b1584d697f1275fe661033c04ad97bd532932eb#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487

File: config/delta-producer/publication-graph-maintainer/config.json [hunk 1]
```diff
@@ -137,5 +137,29 @@
...
+      "useFileDiff": true,
+      "healingInitialBatchSizeInsert": 1000,
+      "updatePublicationGraphSleep": 200,
+      "serveDeltaFiles": true,
+      "deltaInterval": 10000,
+      "deltaPath": "/worship-services-sensitive/delta",
+      "loginPath": "/worship-services-sensitive/login",
+      "publisherUri": "http://data.lblod.info/services/delta-producer-publication-graph-maintainer-ws-sensitive",
+      "filesGraph": "http://redpencil.data.gift/id/deltas/producer/worship-services-sensitive-deltas",
+      "account": "http://data.lblod.info/foaf/account/id/0a3d5be8-6f6b-4c4e-9069-9b7c57a35df3"
```
Comment (Riadabd) [inline config/delta-producer/publication-graph-maintainer/config.json:163]: This should be the right account:
```suggestion
      "account": "http://data.lblod.info/foaf/account/id/f185b63e-5bd9-40a5-8906-4ac1fb5a8db7"
```

If I check in the database locally, it is connected to `http://data.lblod.info/foaf/group/id/c59564df-8a3e-4834-bda4-ceca83e1dad7`, which is the "worship-services-sensitive-deltas" group.

File: config/delta-producer/publication-graph-maintainer/config.json [hunk 2]
```diff
@@ -137,5 +137,29 @@
     "publisherUri": "http://data.lblod.info/services/delta-producer-publication-graph-maintainer-vendor-management",
     "filesGraph": "http://redpencil.data.gift/id/deltas/producer/vendor-management-deltas",
     "account": "http://data.lblod.info/foaf/account/id/0a3d5be8-6f6b-4c4e-9069-9b7c57a35df3"
-  }
+  },
+    "worship-services-sensitive": {
```
Comment (Riadabd) [inline config/delta-producer/publication-graph-maintainer/config.json:141]: The indentation of the "worship-services-sensitive" block should be moved back one tab in order to match with the rest of file.

**Comment by Riadabd** [comment]
> Is there a reason that the route is duplicated ?
> 
> [9b1584d#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487](https://github.com/lblod/app-digitaal-loket/commit/9b1584d697f1275fe661033c04ad97bd532932eb#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487)

Since we will be remove the `/sync/worship-services-sensitive-deltas` bits from the dispatcher configuration, we can maybe update the account URI for `worship-services-sensitive`. At the moment, this file (`config/migrations/2022/20221214000000-worhsip-service-mandate-producer/20221219100000-account.sparql`) defines a group named "worship-services-sensitive-deltas"; we can rename it to "worship-services-sensitive".

File: config/delta-producer/publication-graph-maintainer/config.json [hunk 3]
```diff
@@ -137,5 +137,29 @@
...
+    "skipMuAuthInitialSync": true,
+    "skipMuAuthHealing": true,
+    "useFileDiff": true,
+    "healingInitialBatchSizeInsert": 1000,
+    "updatePublicationGraphSleep": 200,
+    "serveDeltaFiles": true,
+    "deltaInterval": 10000,
+    "deltaPath": "/worship-services-sensitive/delta",
+    "loginPath": "/worship-services-sensitive/login",
+    "publisherUri": "http://data.lblod.info/services/delta-producer-publication-graph-maintainer-ws-sensitive",
```
Comment (Riadabd) [inline config/delta-producer/publication-graph-maintainer/config.json:161]: I was thinking of renaming this line to:
```suggestion
    "publisherUri": "http://data.lblod.info/services/delta-producer-publication-graph-maintainer-worship-services-sensitive",
```
to match the format of the other keys.

Comment (Poltergeistz) [reply config/delta-producer/publication-graph-maintainer/config.json:161]: Good catch I've missed that one

**Comment by cecemel** [comment]
> Is there a reason that the route is duplicated ?
> 
> [9b1584d#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487](https://github.com/lblod/app-digitaal-loket/commit/9b1584d697f1275fe661033c04ad97bd532932eb#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487)

Carful with that; if you change the API you might break consumers. I'll put it in the release notes  that you'll have to verify both config of OP and worship-positions

**Comment by cecemel** [comment]
> > Is there a reason that the route is duplicated ?
> > [9b1584d#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487](https://github.com/lblod/app-digitaal-loket/commit/9b1584d697f1275fe661033c04ad97bd532932eb#diff-1995341ffa86689e960b608816f8937a1dca9622f07c0f5f9801338ac8043e1cL451-R487)
> 
> Since we will be remove the `/sync/worship-services-sensitive-deltas` bits from the dispatcher configuration, we can maybe update the account URI for `worship-services-sensitive`. At the moment, this file (`config/migrations/2022/20221214000000-worhsip-service-mandate-producer/20221219100000-account.sparql`) defines a group named "worship-services-sensitive-deltas"; we can rename it through a migration to "worship-services-sensitive".

So, doing this breaks assumptions in mu-auth.
The groupname is used as postfix of the graph where the data is stored.
https://github.com/lblod/app-digitaal-loket/pull/552/files#diff-4ec8cb33c59c3f9e2a7bbe4dd64369b6f52e4fc783cd883dc7f7ecc410fdd77cR1

I'll revert this change (by adding a new migration)

**Review by Riadabd — APPROVED** [review]
Tested on `app-organization-portal` and `app-worship-database`, and deltas are working as expected :+1: 

Before merging, can you remove the two migrations?

**Pull request merged**